### DISCUSSION TITLE
Ideas for multi player singing game modes

### DISCUSSION THREAD

**Discussion by twollgam** [performous/performous#640] [Ideas]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

I would like to implement some multi player game modes. So this issue is for collecting and discussing ideas.

**Comment by Baklap4** [comment]
What kind of multi-player modes?
As of currently singing, guitar hero are already multiplayer?

**Comment by twollgam** [comment]
Oh sorry, I will be more concrete. In the original Singstar there are modes like "pass the micro". I think  those modes are not implemented actual.

**Comment by Baklap4** [comment]
### Singstar Party mode

#### Best of 3

1. Battle it out across 3 songs: Highest total score wins
2. Two different modes -> One vs One OR Battle
    a. I'm not sure what the difference between 1v1 vs Battle is.
3. After mode selection pick 3 songs and sing the songs.
4. Max totalscore can be 30k (10k per song)

#### Pass the mic

1. Pick a song
2. At semirandom times the game tells you to switch the mic to another player on which they continue with the song.
3. Max score: 10k

Note:
Needs a list of players binded to a certain microphone OR a 'team' concept where the playername is actual the team name.

### Guitar Hero modes

#### Carreer mode

Play 6 songs after eachother get scores after that. If the crowd doesn't like what you're playing you'll get penalties and in the worst case scenario be taken off the stage, failing the game.
Can either be single or coop

#### Face Off mode

Duel between two players on possible different difficulties
Each player plays different parts of the song

#### Pro Face-Off

Same as Face Off but players play at the same difficulty and gotta play the whole song instead of parts.

### Dance Dance Revolution modes

Not sure about 'official dancemodes' but i've asked a few people from that community and they suggested:
- Things like https://www.youtube.com/watch?v=iuT9hBqR-3s
- Lyrics but the lyrics are telling you to do certain stuff while keeping the rhythm/steps
- Meme-charts
- 2 player charts where you do a dance off between the two players
- Story mode?

### Issues related to this

- [ ] Player screen management (https://github.com/performous/performous/issues/574)

**Reply by ooshlablu** [reply]
For Guitar Hero Career mode, I think we'd need a couple of things:
- The Ability to create players, and have them be remembered
- The ability to create bands with said players, have the total band score remembered

Nice to have:
- Some way to create and save playlists so that "Tours" and "Events" can be created that need to be completed throughout a career

**Reply by Baklap4** [reply]
> The Ability to create players, and have them be remembered

There are 2 items for players at the moment.
- [ ] #574 
- [ ] #538 

> Some way to create and save playlists so that "Tours" and "Events" can be created that need to be completed throughout a career

In a way this the same functionality we already have for singing right?
Whilst the guitar hero cd's come with predefined Tours and Events.. we should let the player create these and save them i think

caching (and in a way saving) a playlist was proposed sometime ago -> #406 

Once enough comments on here i'll start creating a Task list so we can work on that :)

**Reply by ooshlablu** [reply]
"Whilst the guitar hero cd's come with predefined Tours and Events.. we should let the player create these and save them i think"

^^ Yes, this is what I meant :-) 
I don't even know if it's possible to extract the Tours and Events from GH or RB

**Comment by twollgam** [comment]
Idea of a singing party mode:
### Shoot out
1. Select players or groups
2. Select n songs. n = number of players/groups * 3.
3. Calculate a random player/group sequence.
4. Calculate a random start song.
5. Sing and Score.
6. The player choose the  song. for the next player/group.
7. Until songs left loop to 5.
8. Winner is the player/group with the highest score.
